src/Tools/dotnet-monitor/CollectionRules/CollectionRuleService.cs

```diff
@@ -30,17 +33,31 @@ CollectionRulesConfigurationProvider provider
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _provider = provider ?? throw new ArgumentNullException(nameof(provider));
             _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+
+            CancellationToken disposalToken = _disposalSource.Token;
+            _reapplyRulesTask = Task.Run(() => ReapplyRulesAsync(disposalToken), disposalToken).SafeAwait();
```

Comment:
Consider pushing the execution of this task into IHostedService instead of the ctor here.